src/components/NetworkDetector/NetworkDetector.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from "react";
+import styled from "styled-components";
+import { ThemeProvider } from "styled-components";
+import { CustomColors } from "../../colors/Colors";
+import { GOOGLE } from "../../configs/apiEndpoints";
+
+const StyledOffline = styled.div`
+  height: 10px;
+  width: 10px;
+  background-color: ${(props) => props.theme.colors.red};
+  border-radius: 50%;
+`;
+
+const StyledSpan = styled.span`
+  margin-left: 12px;
+  font-family: "Poppins";
+  font-size: smaller;
+`;
+
+export default function (ComposedComponent: React.FunctionComponent) {
```

Comment:
abych řekl pravdu, nevidím důvod tady mít HOC (high order component), akorát to podle mě přidává komplexitu, nebo je tu nějaký důvod?? 🤔